nwcsafpps_runner/tests/test_pps_hook.py

```diff
@@ -231,7 +235,42 @@ def test_get_granule_duration(self, mandatory_param, filename):
         delta_t = posttroll_message.get_granule_duration()
         self.assertIsInstance(delta_t, timedelta)
 
-        self.assertAlmostEqual(delta_t.total_seconds(), 86.0, places=5)
+        self.assertAlmostEqual(delta_t.total_seconds(), 84.2, places=5)
+
+        metadata['start_time'] = START_TIME2
+        metadata['end_time'] = END_TIME2
+
+        posttroll_message = PostTrollMessage(0, metadata)
+        delta_t = posttroll_message.get_granule_duration()
+
+        self.assertAlmostEqual(delta_t.total_seconds(), 82.8, places=5)
+
+    @patch('nwcsafpps_runner.pps_posttroll_hook.PostTrollMessage.sensor_is_viirs')
+    @patch('nwcsafpps_runner.pps_posttroll_hook.PostTrollMessage.check_metadata_contains_filename')
+    @patch('nwcsafpps_runner.pps_posttroll_hook.PostTrollMessage.check_metadata_contains_mandatory_parameters')
+    def test_is_segment(self, mandatory_param, filename, sensor_is_viirs):
+        """Test the determination of whether data is a segment or not."""
+        from nwcsafpps_runner.pps_posttroll_hook import PostTrollMessage
+
+        mandatory_param.return_value = True
+        filename.return_value = True
+        sensor_is_viirs.return_value = True
+
+        metadata = self.metadata_with_start_and_end_times
+
+        posttroll_message = PostTrollMessage(0, metadata)
+        delta_t = timedelta(seconds=82.8)
+
+        with patch.object(PostTrollMessage, 'get_granule_duration', return_value=delta_t) as mock_method:
+            result = posttroll_message.is_segment()
+            self.assertTrue(result)
```

Comment:
Could you also add a check for 48 scans?

Reply:
I thought I had!?
I have a test for 82.2 seconds (47 scans) and 84.1 sec (48 scans).

The thing is, as far as I interpret things, that in the SDR files you have the start and end time in terms of time of the first scan, and time of the last scan. But to get to the duration of the entire granule you need to add the time length of one scan (~1.8 sec). Thus, the duration of 48 scans is ~85.9 sec and ~84.1 seconds for 47 scans.

Reply:
I don't see the test for 84.1 seconds in `test_is_segment`, what am I missing?

Reply:
https://github.com/adybbroe/pytroll-pps-runner/blob/a0a7441893c4e5436474e009c9d9ea8858376ec3/nwcsafpps_runner/tests/test_pps_hook.py#L262

Reply:
@mraspaud Do you see it now?